### PR TITLE
EPMRPP-40028 || Test case id impl

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package com.epam.reportportal.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java]: Document it.
Describe when and how to use it with TestCaseId

File: src/main/java/com/epam/reportportal/annotations/TestCaseId.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * If value is not provided it will be generated on the Report Portal
+ * back-end side as hash of the generated {@link UniqueID}
+ *
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface TestCaseId {
+
+	int value() default 0;
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/annotations/TestCaseId.java]: document it

File: src/main/java/com/epam/reportportal/annotations/TestCaseId.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	boolean isParameterized() default false;
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/annotations/TestCaseId.java]: document it

Comment (pbortnik) [inline src/main/java/com/epam/reportportal/annotations/TestCaseId.java]: can be just 'parametrized'

File: src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.FIELD, ElementType.PARAMETER })
+public @interface TestCaseIdKey {
+
+	boolean isInteger();
```
Comment (pbortnik) [inline src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java]: maybe better 'number' or resolve it from parameter type?

File: src/main/java/com/epam/reportportal/annotations/TestCaseId.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/TestCaseId.java]: renamed

File: src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java (hunk 4 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java]: renamed to 'numeric'

File: src/main/java/com/epam/reportportal/annotations/TestCaseId.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/TestCaseId.java]: documented

File: src/main/java/com/epam/reportportal/annotations/TestCaseId.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/TestCaseId.java]: done

File: src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/reportportal/annotations/TestCaseIdKey.java]: Documentation provided

**Pull request merged**